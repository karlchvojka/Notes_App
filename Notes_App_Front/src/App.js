import React, { useState, useEffect } from 'react';
import '../styles/_reset.scss';
import '../styles/fonts.scss';
import '../styles/variables.scss';
import './App.scss';

// Component Includes
import Header from './components/molecules/Header';
import SideBar from './components/molecules/SideBar';
import TaskForm from './components/organisims/TaskForm';
import DocsList from './components/molecules/DocsList';

function App() {
  const [notes, setNotes] = useState([]);

  return (
    <div id="App">
      <Header />
      <section className="mainWrap">
        <SideBar notes={notes} />
        <section className="mainInner">
          <TaskForm currNotes={notes} setNotes={setNotes} />
          <DocsList notes={notes} setNotes={setNotes} />
        </section>
      </section>
    </div>
  );
}

export default App;
