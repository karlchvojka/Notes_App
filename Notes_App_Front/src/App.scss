/*
  These styles are some defaults meant for the Template.
  These are default, feel free to change them.

  I would suggest leaving the imports tho ;)
*/

@import '../styles/_reset.scss';
@import '../styles/variables.scss';
@import '../styles/fonts.scss';

body {
  background-color: $darkGrey;
  background-size: 75px 75px;
  background-image:
    linear-gradient(0deg, transparent 24%, rgba(255, 255, 255, .05) 25%, rgba(255, 255, 255, .05) 26%, transparent 27%, transparent 74%, rgba(255, 255, 255, .05) 75%, rgba(255, 255, 255, .05) 76%, transparent 77%, transparent),
    linear-gradient(90deg, transparent 24%, rgba(255, 255, 255, .05) 25%, rgba(255, 255, 255, .05) 26%, transparent 27%, transparent 74%, rgba(255, 255, 255, .05) 75%, rgba(255, 255, 255, .05) 76%, transparent 77%, transparent);
}

header {
  background-color: $headerGrey;
  padding:20px;
  text-align:center;
}
.formWrap {
  max-width: 80%;
  margin: 0 auto;
}
.mainWrap {
  height:100vh;
  max-width:80%;
  margin: 0 auto;
}
