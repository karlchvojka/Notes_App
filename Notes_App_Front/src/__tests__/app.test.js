import React from "react";

import { render, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer';

import App from "../App";

afterEach(cleanup);

describe('when the app renders', () => {

  it("renders without crashing", () => {
    const wrapper = renderer.create(<App />);
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });


});
