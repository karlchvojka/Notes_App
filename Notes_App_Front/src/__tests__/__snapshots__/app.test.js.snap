// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the app renders renders without crashing 1`] = `
<div
  id="App"
>
  <header
    id="mainHeader"
  >
    <h1>
      Burn_Net: Notes
    </h1>
  </header>
  <div
    className="formWrap"
  >
    <form
      id="taskForm"
      onSubmit={[Function]}
    >
      <label>
        <span>
          Title:
        </span>
        <input
          id="noteTitleInput"
          name="titleInput"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label>
        <span>
          Date:
        </span>
        <input
          id="noteDateInput"
          name="dateInput"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label>
        <span>
          Category:
        </span>
        <input
          id="noteCatInput"
          name="catInput"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <textarea
        id="noteContentInput"
        name="contInput"
        onChange={[Function]}
        type="textArea"
        value=""
      />
      <button
        id="submitButton"
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
  <div
    className="mainWrap"
  >
    <ul />
  </div>
</div>
`;
